libmscore/noteentry.cpp

```diff
@@ -296,6 +298,14 @@ void Score::putNote(const QPointF& pos, bool replace, bool insert)
       Score* score = p.segment->score();
       // it is not safe to call Score::repitchNote() if p is on a TAB staff
       bool isTablature = staff(p.staffIdx)->isTabStaff(p.segment->tick());
+
+      //  adjust pos to staff type change (test with other input methods!)
+      Staff* ss = score->staff(p.staffIdx);
+      int stepOffset = ss->staffType(p.segment->tick())->stepOffset();
+      qreal stYOffset = ss->staffType(p.segment->tick())->yoffset().val();
+      qreal lineDist = ss->staffType(p.segment->tick())->lineDistance().val();
+      p.line -= stepOffset + 2 * stYOffset / lineDist;
+
       if (score->inputState().usingNoteEntryMethod(NoteEntryMethod::REPITCH) && !isTablature)
```

Comment:
This code recalculates the click position on a modified staff so the note is added to the intended (clicked) line or space.  (Testing  for any other input method that implies clicking or could be affected by this code.)